### PR TITLE
OCM-CLI: Support BYO-VPC cluster creation flags

### PR DESCRIPTION
**Pull request by zvikorn**
Signed-off-by: Tzvika Avni <tavni@redhat.com>

https://issues.redhat.com/browse/SDA-5116

This code addition has more or less the same behavior as ROSA does. Code is very similar

@tzvatot @zgalor @adlerOri

### CODE REVIEW

File: cmd/ocm/create/cluster/cmd.go [hunk 1]
```diff
@@ -61,6 +63,7 @@ var args struct {
 	computeMachineType string
 	computeNodes       int
 	autoscaling        c.Autoscaling
+	byovpc c.BYOVPC
```
Comment (tzvatot) [inline cmd/ocm/create/cluster/cmd.go:66]: indentation issue

Comment (tzvatot) [reply cmd/ocm/create/cluster/cmd.go:66]: byoVPC?

File: cmd/ocm/create/cluster/cmd.go [hunk 2]
```diff
@@ -507,6 +529,96 @@ func promptName(argv []string) error {
 	return fmt.Errorf("A cluster name must be specified")
 }
 
+func promptBYOVPC(fs *pflag.FlagSet, connection *sdk.Connection, cmd *cobra.Command) error {
+	providedSubnetIDs := strings.Split(args.byovpc.SubnetIDs, ",")
+	areSubnetsProvided := len(args.byovpc.SubnetIDs) > 0
+	if !args.byovpc.Enabled && args.ccs.Enabled && !areSubnetsProvided && args.interactive {
```
Comment (tzvatot) [inline cmd/ocm/create/cluster/cmd.go:535]: Decompose conditional

File: cmd/ocm/create/cluster/cmd.go (hunk 1 above)
Comment (zvikorn) [reply cmd/ocm/create/cluster/cmd.go:66]: must be exported like other cluster fields

**Comment by zvikorn** [comment]
@tzvatot @LeshemNoa @adlerOri PTAL

**Review by jhernand — APPROVED** [review]
Looks good, thanks @zvikorn.

**Pull request merged**